fem/tmop.hpp

```diff
@@ -860,9 +879,14 @@ class TargetConstructor
        nodes are used by all target types except IDEAL_SHAPE_UNIT_SIZE. */
    void SetNodes(const GridFunction &n) { nodes = &n; avg_volume = 0.0; }
 
+   /** @brief Get the nodes to be used in the target-matrix construction. */
+   const GridFunction *GetNodes() const { return nodes; }
+
    /// Used by target type IDEAL_SHAPE_EQUAL_SIZE. The default volume scale is 1.
    void SetVolumeScale(double vol_scale) { volume_scale = vol_scale; }
 
+   const TargetType &Type() const { return target_type; }
```

Comment:
`GetTargetType` seems like a better name to me. Also, returning an enum type by reference seems odd. Overall, I'd replace this with
```c++
   TargetType GetTargetType() const { return target_type; }
```